roles/nvidia-docker/tasks/main.yml

```diff
@@ -0,0 +1,73 @@
+- name: "Install GCCv10"
+  package:
+    name: gcc10.x86_64
+    state: present
+
+- name: "Remove nouveau kernel module"
+  modprobe:
+    name: nouveau
+    state: absent
+
+- name: "Disable nouveau kernel module"
+  copy:
+    dest: /usr/lib/modprobe.d/nvidia-installer-disable-nouveau.conf
+    content: |
+      blacklist nouveau
+      options nouveau modeset=0
+
+- name: "Download nvidia-{{ nvidia_driver_version }} driver"
+  get_url:
+    url: "https://us.download.nvidia.com/XFree86/Linux-x86_64/{{ nvidia_driver_version }}/NVIDIA-Linux-x86_64-{{ nvidia_driver_version }}.run"
+    dest: "/root/NVIDIA-{{ nvidia_driver_version }}.run"
+
+- name: "Install nvidia-{{ nvidia_driver_version }} driver"
+  shell: "sh /root/NVIDIA-{{ nvidia_driver_version }}.run --silent"
```

Comment:
It looks like the NVIDIA driver takes some advanced options that might help us when running in a chroot with a different kernel:

```
  -k KERNEL-NAME, --kernel-name=KERNEL-NAME
      Build and install the NVIDIA kernel module for the non-running kernel specified by KERNEL-NAME (KERNEL-NAME should be the output of `uname -r` when the target kernel is actually running).  This option implies '--no-precompiled-interface'.  If the options '--kernel-install-path' and '--kernel-source-path' are not given, then they will be inferred from KERNEL-NAME; eg: '/lib/modules/KERNEL-NAME/kernel/drivers/video/' and '/lib/modules/KERNEL-NAME/build/', respectively.
  --dkms
      nvidia-installer can optionally register the NVIDIA kernel module sources, if installed, with DKMS, then build and install a kernel module using the DKMS-registered sources.  This will allow the DKMS infrastructure to automatically build a new kernel module when changing kernels.  During installation, if DKMS is detected, nvidia-installer will ask the user if they wish to register the module with DKMS; the default response is 'no'.  This option will bypass the detection of DKMS, and cause the installer to attempt a DKMS-based installation regardless of whether DKMS is present.
```

The `--dkms` option will ensure that the module gets rebuilt after a new kernel is installed, which is useful since our systems auto-upgrade.

We can use the `-k` option to tell it to use the kernel that was installed by kernel-ml.
You can find the version of the kernel the system will use on next boot by running `grubby --default-kernel`, which should output something like:
```
/boot/vmlinuz-5.16.13-1.el7.elrepo.x86_64
```

The installer also has some options that would be useful to limit the cruft that gets installed:

```
  --no-wine-files
      Do not install any of the Wine driver files.
```